### ISSUE TITLE
High CPU usage by TypingsInstaller

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#18263]
_From @oliversalzburg on August 29, 2017 8:59_

- VSCode Version: Code 1.15.1 (41abd21afdf7424c89319ee7cb0445cc6f376959, 2017-08-16T18:07:25.676Z)
- OS Version: Windows_NT x64 10.0.15063
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|1.2.11
pgsql|dou|0.0.8
EditorConfig|Edi|0.9.4
JSDocTagComplete|Hoo|0.0.2
vscode-esformatter|how|1.5.2
output-colorizer|IBM|0.1.2
vscode-puglint|mrm|2.3.0
code-spell-checker|str|1.3.7

(1 theme extensions excluded)

---

Opening certain projects causes a process to spawn that apparently downloads TS support files. These processes regularly cause extremely high CPU usage over long periods of time. I don't think this is intentional.

The log has a lot of:
```
Response has been sent.
Closing file watchers for project '/dev/null/inferredProject3*'
Closing file watchers for project '/dev/null/inferredProject3*' - done.
Installing typings ["emitter"]
Entry for package 'emitter' does not exist in local types registry - skipping...
All typings are known to be missing or invalid - no need to go any further
```

I'm also seeing this, which looks like an error:
```
Installing typings ["@fairmanager/core-secur","emitter"]
Package '@fairmanager/core-secur' is scoped and currently is not supported
...
Installing typings ["@fairmanager/core-securi","emitter"]
Package '@fairmanager/core-securi' is scoped and currently is not supported
...
Installing typings ["@fairmanager/core-security","emitter"]
Package '@fairmanager/core-security' is scoped and currently is not supported
```
The package in question is named `core-security2`.

_Copied from original issue: Microsoft/vscode#33339_

**Comment by mjbvz** [comment]
@oliversalzburg,  can you please try this out on the [latest VSCode insiders](https://code.visualstudio.com/insiders) which includes TS 2.5.2. I believe we added some additional logging around the typings installer

**Comment by mhegazy** [comment]
the log looks fine to me. the server will go through list of modules and try to find ones to install. this includes ones it knows do not exist on `@types` and ones that are scoped packages. so these log entries are fine.

can you share the tsserver log, and the full typing installer log for us to look further?

**Comment by oliversalzburg** [comment]
@mhegazy Performing a search for only a partial package name does not read like an application performing correctly.

So what do you guys want me to do now? Try Insiders? Collect logs? Both?

**Comment by mhegazy** [comment]
> Performing a search for only a partial package name does not read like an application performing correctly.

so you are saying `"@fairmanager/core-security"` is not the correct name? can you share more context.

> So what do you guys want me to do now? Try Insiders? Collect logs? Both?

both would be ideal. thanks.

**Comment by oliversalzburg** [comment]
That's right. The package name has another `2` at the end. But I was more concerned about it listing, for example, `@fairmanager/core-secu` in the log.

Regardless, I installed Insiders and the behavior is very different. First of all, all my projects are opened in one workspace, due to the new workspace feature, which is very nice :) Second, there is little to no CPU usage.

So either there is something specific about my plugins or configuration of my stable VS Code, or the issue was fixed.

I have copied my user settings to Insiders, but that did not cause any issues. Is there a quick way to install the same extensions in Insiders that I use in the stable build?
